### PR TITLE
d/aws_networkmanager_core_policy_document

### LINKED ISSUE #23782: resource/aws_networkmanager_core_network
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

support AWS CloudWAN `core network` feature.

### New or Affected Resource(s)

* aws_networkmanager_core_network
* aws_networkmanager_core_network_policy
* data.aws_networkmanager_core_network_policy_document

* aws_networkmanager_vpc_attachment
* aws_networkmanager_accept_attachment
* aws_networkmanager_core_network_live_policy_version

### Potential Terraform Configuration

```hcl
resource "aws_networkmanager_core_network" "cloudwan" {
  global_network_id = <>
}

resource "aws_networkmanager_core_network_policy" "cloudwan" {
  core_network_id = aws_networkmanager_core_network.cloudwan.id
  policy_document = data.aws_networkmanager_core_network_policy_document.cloudwan.json
}

data "aws_networkmanager_core_network_policy_document" "cloudwan" {
  # https://docs.aws.amazon.com/vpc/latest/cloudwan/cloudwan-policies-json.html
}

resource "aws_networkmanager_core_network_live_policy_version" "live" {
  core_network_id = aws_networkmanager_core_network.cloudwan.id
  policy_version_id = aws_networkmanager_core_network_policy.cloudwan.id
}
```

Request and accept attachment

```hcl
resource "aws_networkmanager_vpc_attachment" "cloudwan" {
  core_network_id = aws_networkmanager_core_network.cloudwan.id
  vpc_arn = ""
  subnet_arns = [""]
}

resource "aws_networkmanager_accept_attachment" "cloudwan" {
  attachment_id = aws_networkmanager_vpc_attachment.cloudwan.id
}
```

### References

* https://docs.aws.amazon.com/cli/latest/reference/networkmanager/create-core-network.html
* https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-aws-cloud-wan-preview/
* https://docs.aws.amazon.com/cli/latest/reference/networkmanager/create-vpc-attachment.html
* https://docs.aws.amazon.com/cli/latest/reference/networkmanager/accept-attachment.html
* https://docs.aws.amazon.com/vpc/latest/cloudwan/cloudwan-policies-json.html

### PR DESCRIPTION
**Pull request by drewmullen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Partially Closes #23782

PR also includes a new guide: `"Using Terraform awscc provider with aws provider"`
**TODO** Open a new PR to update the link once the new guide has been published.

Output from acceptance testing:

```
$ make testacc TESTS=TestAccCoreNetworkPolicyDocumentDataSource_basic PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccCoreNetworkPolicyDocumentDataSource_basic'  -timeout 180m
=== RUN   TestAccCoreNetworkPolicyDocumentDataSource_basic
=== PAUSE TestAccCoreNetworkPolicyDocumentDataSource_basic
=== CONT  TestAccCoreNetworkPolicyDocumentDataSource_basic
--- PASS: TestAccCoreNetworkPolicyDocumentDataSource_basic (13.61s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     18.615s
```

### CODE REVIEW

File: website/docs/guides/using-aws-with-awscc-provider.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+subcategory: ""
+layout: "aws"
+page_title: "Using Terraform awscc provider with aws provider"
```
Comment (maryelizbeth) [inline website/docs/guides/using-aws-with-awscc-provider.md:4]:
```suggestion
page_title: "Using the Terraform awscc provider with aws provider"
```

File: website/docs/guides/using-aws-with-awscc-provider.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+description: |-
+  Managing resource tags with the Terraform AWS Provider.
+---
+
+# Using AWS & AWSCC Provider Together
+
+~> **NOTE**: Please note the AWSCC provider is currently in technical preview. This means some aspects of its design and implementation are not yet considered stable. We are actively looking for community feedback in order to solidify its form.
```
Comment (maryelizbeth) [inline website/docs/guides/using-aws-with-awscc-provider.md:11]:
```suggestion
~> **NOTE**: The `awscc` provider is currently in technical preview. This means some aspects of its design and implementation are not yet considered stable for production use. We are actively looking for community feedback in order to identify needed improvements.
```

File: website/docs/guides/using-aws-with-awscc-provider.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+The [HashiCorp Terraform AWS Cloud Control Provider](https://registry.terraform.io/providers/hashicorp/awscc/latest) aims to bring Amazon Web Services (AWS) resources to Terraform users faster. The new provider is automatically generated, which means new features and services on AWS can be supported right away. The AWS Cloud Control provider supports hundreds of AWS resources, with more support being added as AWS service teams adopt the Cloud Control API standard.
+
+For Terraform users managing infrastructure on AWS, we expect the AWSCC provider will be used alongside the existing AWS provider. This guide is provided to show guidance and an example of using the providers together to deploy an AWS Cloud WAN Core Network.
```
Comment (maryelizbeth) [inline website/docs/guides/using-aws-with-awscc-provider.md:15]:
```suggestion
For Terraform users managing infrastructure on AWS, we expect the AWSCC provider will be used alongside the existing AWS provider. This guide is an example of how you might use the providers together to deploy an AWS Cloud WAN Core Network.
```

File: website/docs/guides/using-aws-with-awscc-provider.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+<!-- /TOC -->
+
+## AWS Cloud Wan
+
+In this guide we will deploy [AWS Cloud WAN](https://aws.amazon.com/cloud-wan/) to demonstrate how both AWS & AWSCC can work togther. Cloud WAN is a wide area networking (WAN) service that helps you build, manage, and monitor a unified global network that manages traffic running between resources in your cloud and on-premises environments.
+
+With Cloud WAN, you define network policies that are used to create a global network that spans multiple locations and networks—eliminating the need to configure and manage different networks individually using different technologies. Your network policies can be used to specify which of your Amazon Virtual Private Clouds (VPCs) and on-premises locations you wish to connect through AWS VPN or third-party software-defined WAN (SD-WAN) products, and the Cloud WAN central dashboard generates a complete view of the network to monitor network health, security, and performance. Cloud WAN automatically creates a global network across AWS Regions using Border Gateway Protocol (BGP) so you can easily exchange routes around the world.
```
Comment (maryelizbeth) [inline website/docs/guides/using-aws-with-awscc-provider.md:32]:
```suggestion
With Cloud WAN, you define network policies that are used to create a global network that spans multiple locations and networks—eliminating the need to configure and manage different networks individually using different technologies. Your network policies can be used to specify which of your Amazon Virtual Private Clouds (VPCs) and on-premises locations you wish to connect through AWS VPN or third-party software-defined WAN (SD-WAN) products, and the Cloud WAN central dashboard generates a complete view of the network to monitor network health, security, and performance. Cloud WAN automatically creates a global network across AWS Regions using Border Gateway Protocol (BGP), so you can easily exchange routes around the world.
```

This is a tiny one, feel free to disregard!

File: website/docs/guides/using-aws-with-awscc-provider.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+For more information on AWS Cloud WAN see [the documentation.](https://docs.aws.amazon.com/vpc/latest/cloudwan/what-is-cloudwan.html)
+
+## Specifying Multiple Providers
+
+Terraform can use many providers by specifying them in your `terraform` configuration block:
```
Comment (maryelizbeth) [inline website/docs/guides/using-aws-with-awscc-provider.md:38]:
```suggestion
Terraform can use many providers at once, as long as they are specified in your `terraform` configuration block:
```

**Review by maryelizbeth — COMMENTED** [review]
Left a few suggestions! Thanks for putting this together, @drewmullen!

**Comment by drewmullen** [comment]
terrafmt appears to have nothing to do with my code:
```
Run # efs/file_system_test.go: argument name is format verb and replaced with quoted string. https://github.com/katbyte/terrafmt/issues/47
  # efs/file_system_test.go: argument name is format verb and replaced with quoted string. https://github.com/katbyte/terrafmt/issues/47
  # kms/grant_test.go: argument name is format verb and replaced with quoted string. https://github.com/katbyte/terrafmt/issues/47
  # quicksight/user_test.go: format verb as resource name (%[1]q). https://github.com/katbyte/terrafmt/issues/[4](https://github.com/hashicorp/terraform-provider-aws/runs/6237338102?check_suite_focus=true#step:7:4)8
  # sns/platform_application_test.go: argument name is format verb and replaced with quoted string. https://github.com/katbyte/terrafmt/issues/47
  find ./internal/service -type f -name '*_test.go' \
    | sort -u \
    | grep -v efs/file_system_test.go \
    | grep -v kms/grant_test.go \
    | grep -v quicksight/user_test.go \
    | grep -v sns/platform_application_test.go \
    | xargs -I {} terrafmt diff --check --fmtcompat {}
```

**Comment by drewmullen** [comment]
Guessing the markdownlint issue has to do with my [stubbed](https://github.com/hashicorp/terraform-provider-aws/pull/24368/files#diff-906c12fbc2c1855e038b3ea570acce7e32597721ac3b50fbe762a3863e24eadfR15) `<change-me>`. Just need to know what URL the new guide will be at then i can update that URL...

File: internal/service/networkmanager/core_network_policy_document_data_source.go [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+			k["key"] = true
+			condition.Key = key.(string)
+		}
+		if v := cfgCond["value"]; v != "" {
+			k["value"] = true
+			condition.Value = v.(string)
+		}
+
+		if t == "any" {
+			for _, key := range k {
```
Comment (GlennChia) [inline internal/service/networkmanager/core_network_policy_document_data_source.go:489]: Nice implementation!

File: internal/service/networkmanager/core_network_policy_model.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+	SegmentActions           []*CoreNetworkPolicySegmentAction          `json:"segment-actions,omitempty"`
+}
+
+type CoreNetworkPolicySegmentAction struct {
+	Action                string      `json:"action"`
+	Destinations          interface{} `json:"destinations,omitempty"`
+	DestinationCidrBlocks interface{} `json:"destination-cidr-blocks,omitempty"`
+	Mode                  string      `json:"mode,omitempty"`
+	Segment               string      `json:"segment,omitempty"`
+
```
Comment (GlennChia) [inline internal/service/networkmanager/core_network_policy_model.go:22]: style: extra line here

File: website/docs/d/networkmanager_core_network_policy_document.html.markdown [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+* `type` - Valid values include: `account-id`, `any`, `tag-value`, `tag-exists`, `resource-id`, `region`, `attachment-type`.
+* `operator` - Valid values include: `equals`, `not-equals`, `contains`, `begins-with`.
+* `key` - string value
+* `value` - string value
+
+### `core_network_configuration`
+
+The following arguments are available:
+
+* `asn_ranges` - List of strings containing Autonomous System Numbers (ASNs) to assign to Core Network Edges. By default, the core network automatically assigns an ASN for each Core Network Edge but you can optionally define the ASN in the edge-locations for each Region. The ASN uses an array of integer ranges only from 64512 to 65534 and 4200000000 to 4294967294. No other ASN ranges can be used.
```
Comment (GlennChia) [inline website/docs/d/networkmanager_core_network_policy_document.html.markdown:208]: We can change "64512" to "`64512`" for readability. Same for the other numbers for this argument, to be consistent with how other parts of the documentation have them highlighted

File: website/docs/d/networkmanager_core_network_policy_document.html.markdown [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+      "segment": "shared",
+      "share-with": "*"
+    }
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are optional:
```
Comment (GlennChia) [inline website/docs/d/networkmanager_core_network_policy_document.html.markdown:167]: From the Schema code it seems `core_network_configuration` is set to `Required: true,`. Hence, not all of the arguments are Optional?

File: website/docs/d/networkmanager_core_network_policy_document.html.markdown [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+* `segment_actions` (Optional) - A block argument, `segment_actions` define how routing works between segments. By default, attachments can only communicate with other attachments in the same segment. Detailed below.
+
+### `attachment_policies`
+
+The following arguments are available:
+
+* `action` - The action to take when a condition is true. Detailed Below.
+* `condition_logic` (Optional) - Valid values include `and` or `or`. This is a mandatory parameter only if you have more than one condition. The `condition_logic` apply to all of the conditions for a rule, which also means nested conditions of and or or are not supported. Use `or` if you want to associate the attachment with the segment by either the segment name or attachment tag value, or by the chosen conditions. Use `and` if you want to associate the attachment with the segment by either the segment name or attachment tag value and by the chosen conditions. Detailed Below.
```
[the remark this answers is not in the document]
Comment (drewmullen) [reply website/docs/d/networkmanager_core_network_policy_document.html.markdown:179]: good catch! thank you!

File: internal/service/networkmanager/core_network_policy_document_data_source.go [hunk 11]
```diff
@@ -0,0 +1,643 @@
...
+			"core_network_configuration": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"asn_ranges": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
```
[the remark this answers is not in the document]
Comment (drewmullen) [reply internal/service/networkmanager/core_network_policy_document_data_source.go:134]: I believe that documentation is either incorrect or can accept either string or int. here is an example of how its passed via json:

```json
    "asn-ranges": [
      "64512-65534",
      "4200000000-4294967294"
    ],
```

File: website/docs/d/networkmanager_core_network_policy_document.html.markdown (hunk 9 above)
Comment (drewmullen) [reply website/docs/d/networkmanager_core_network_policy_document.html.markdown:167]: good catch! im changing to: `The following arguments are available:`

**Comment by GlennChia** [comment]
Changes LGTM!

File: internal/service/networkmanager/core_network_policy_document_data_source.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (drewmullen) [reply internal/service/networkmanager/core_network_policy_document_data_source.go:134]: i sent a change request update to the official docs using the `feedback` button!

File: internal/service/networkmanager/core_network_policy_document_data_source_test.go [hunk 12]
```diff
@@ -0,0 +1,505 @@
...
+    action {
+      association_method = "constant"
+      segment            = "GoodSegmentSpecification"
+      require_acceptance = true
+    }
+  }
+}
+`
+
+//lintignore:AWSAT003
```
[the remark this answers is not in the document]
Comment (drewmullen) [reply internal/service/networkmanager/core_network_policy_document_data_source_test.go:274]: I'll leave this for now, if anyone at Hashi has a preference please lmk

[the remark this answers is not in the document]
Comment (ewbankkit) [reply internal/service/networkmanager/core_network_policy_document_data_source_test.go:274]: Let's leave it. If it isn't required then one of the linters should complain.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccCoreNetworkPolicyDocumentDataSource_ PKG=networkmanager
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/networkmanager/... -v -count 1 -parallel 20 -run='TestAccCoreNetworkPolicyDocumentDataSource_'  -timeout 180m
=== RUN   TestAccCoreNetworkPolicyDocumentDataSource_basic
=== PAUSE TestAccCoreNetworkPolicyDocumentDataSource_basic
=== CONT  TestAccCoreNetworkPolicyDocumentDataSource_basic
--- PASS: TestAccCoreNetworkPolicyDocumentDataSource_basic (16.86s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager	23.805s
```

**Comment by ewbankkit** [comment]
New CloudFormation Resource Schema definitions and generated Terraform resources/data sources: https://github.com/hashicorp/terraform-provider-awscc/pull/497.

File: website/docs/d/networkmanager_core_network_policy_document.html.markdown [hunk 13]
```diff
@@ -12,7 +12,7 @@ Generates a Core Network policy document in JSON format for use with resources t
 
 Using this data source to generate policy documents is *optional*. It is also valid to use literal JSON strings in your configuration or to use the `file` interpolation function to read a raw JSON policy document from a file.
 
--> For more information about building AWS Core Network policy documents with Terraform, see the [Using AWS & AWSCC Provider Together Guide](https://learn.hashicorp.com/terraform/aws/<changeme>).
+-> For more information about building AWS Core Network policy documents with Terraform, see the [Using AWS & AWSCC Provider Together Guide](https://github.com/hashicorp/terraform-provider-aws/tree/main/website/docs/guides/using-aws-awscc-together).
```
Comment (ewbankkit) [inline website/docs/d/networkmanager_core_network_policy_document.html.markdown:15]: I don't think this is going to work as there is no such file.
Should it be https://learn.hashicorp.com/tutorials/terraform/using-aws-awscc-together?

Comment (drewmullen) [reply website/docs/d/networkmanager_core_network_policy_document.html.markdown]: ack - i see what she meant now :D

**Comment by ewbankkit** [comment]
@drewmullen Thanks for the contribution :tada: :clap:.

**Pull request merged**